src/python/WMCore/WMSpec/Steps/Executors/LogArchive.py

```diff
@@ -58,9 +57,22 @@ def execute(self, emulator=None, **overrides):
             return emulator.emulate(self.step, self.job)
 
         overrides = {}
+        #TODO need to set override using addOverride method in WMStep
         if hasattr(self.step, 'override'):
             overrides = self.step.override.dictionary_()
 
+        eosStageOutParams = {}
+        eosStageOutParams['command'] = "xrdcp"
+        eosStageOutParams['option'] = "--wma-disablewriterecovery"
+        eosStageOutParams['phedex-node'] = overrides.get('eos-phedex-node', "T2_CH_CERN")
+        eosStageOutParams['lfn-prefix'] = overrides.get('eos-lfn-prefix',
+                                            "root://eoscms.cern.ch//eos/cms/store/logs/prod/recent")
+
+        if eosStageOutParams['lfn-prefix']:
+            setEOSLog = True
```

Comment:
honestly, I don't like it :) setEOSLog is always True anyways. This will be - by default - enabled in new agents, so why "control" whether we upload it to EOS or not.
As you said, it's ugly and make the code more unreadable :)

Reply:
I know, but I think we discussed that we can give the hook if we don't want to transfer the file to eos for some reason. (i.e. test workflows, etc). The we we can overwrite overwriteParam value to None. so we don't put unnecessary logs there.

Reply:
If you agree this control is necessary. I will beautify the code. 😄